homeassistant/components/ozw/services.py

```diff
@@ -37,6 +44,103 @@ def async_register(self):
             ),
         )
 
+        self._hass.services.async_register(
+            LOCK_DOMAIN,
+            const.SERVICE_SET_USERCODE,
+            self.set_usercode,
+            schema=vol.Schema(
+                {
+                    vol.Optional(const.ATTR_INSTANCE_ID, default=1): vol.Coerce(int),
+                    vol.Required(const.ATTR_NODE_ID): vol.Coerce(int),
+                    vol.Required(const.ATTR_CODE_SLOT): vol.Coerce(int),
+                    vol.Required(const.ATTR_USERCODE): cv.string,
+                }
+            ),
+        )
+
+        self._hass.services.async_register(
+            LOCK_DOMAIN,
+            const.SERVICE_GET_USERCODE,
+            self.get_usercode,
+            schema=vol.Schema(
+                {
+                    vol.Optional(const.ATTR_INSTANCE_ID, default=1): vol.Coerce(int),
+                    vol.Required(const.ATTR_NODE_ID): vol.Coerce(int),
+                    vol.Required(const.ATTR_CODE_SLOT): vol.Coerce(int),
+                }
+            ),
+        )
+
+        self._hass.services.async_register(
+            LOCK_DOMAIN,
+            const.SERVICE_CLEAR_USERCODE,
+            self.clear_usercode,
+            schema=vol.Schema(
+                {
+                    vol.Optional(const.ATTR_INSTANCE_ID, default=1): vol.Coerce(int),
+                    vol.Required(const.ATTR_NODE_ID): vol.Coerce(int),
+                    vol.Required(const.ATTR_CODE_SLOT): vol.Coerce(int),
+                }
+            ),
+        )
+
+    def set_usercode(self, service):
```

Comment:
Maybe these services should be entity services instead? Then the user doesn't need to know the instance and node id.

Reply:
The main goal was to attempt to bring parity with the service calls from zwave. Minus the instance_id of course. These calls are integral in all the user code managing scripts/packages used by Home Assistant users.

Reply:
There's also those edge cases of users who may be running 2 instances of OpenZwave as well.

Reply:
By linking it to entity IDs, that is all taking care of plus users don't need to know the instance and node IDs. 

What are these services doing? I think that these need to be admin services.

Reply:
Resolved this in the latest commit.